---
title: "MS_Design_Explorer"
date: "7/6/2021"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
options(stringsAsFactors = F)
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(janitor)
library(httr)
```


# Retrieve encode sample report here:

https://www.encodeproject.org/report/?type=Experiment&status=released&replicates.library.biosample.donor.organism.scientific_name=Homo+sapiens&biosample_ontology.term_name=heart+right+ventricle&biosample_ontology.term_name=heart&biosample_ontology.term_name=heart+left+ventricle&assay_slims=DNA+binding&biosample_ontology.term_name=mesodermal+cell&biosample_ontology.term_name=cardiovascular+progenitor+cell&biosample_ontology.term_name=cardiac+myoblast&biosample_ontology.term_name=cardiac+muscle+cell&biosample_ontology.classification!=primary+cell&biosample_ontology.term_name=H7&assay_slims=Transcription&assay_title!=RAMPAGE&assay_title!=RNA+microarray&assay_title!=microRNA-seq&assay_title!=microRNA+counts&assay_title!=small+RNA-seq&assay_title!=long+read+RNA-seq&assay_slims=DNA+accessibility&assay_title!=DNase-seq&assay_title!=ATAC-seq&assay_title!=total+RNA-seq&assay_title!=polyA+plus+RNA-seq


Here we are going to use our ENCODE.TSV report to make a design file for 
Nextflow ChiP-seeker. 

For the input design file Nextflow needs a few things:

group	replicate	fastq	antibody	control

* group is the DBP and controls listed individually
* replicate indicates which replicate of the given DBP 
* Antibody same for all DBPs but input controls are blank
* Control tells which "group" are the inputs.


Now we start the hard part of coding such a simple sounding design file :)


We are going to use the ENCODE API instead of the web interface. Some information
cannot be retrieved such as MD5sum and other aspects in the .TSV

You can see all the information you can gather using the API and what the code
would look like here:

https://app.swaggerhub.com/apis-docs/encodeproject/api/basic_search#/Experiment

Notice the URL produced above has a lot of the information -- so we can use the 
API to tell it exactly what we want in code rather than clicking buttons --


# Running functions to get fastq info
```{R}
contstruct_query <- function(experiment_accession,
                             base_url = "https://www.encodeproject.org/report.tsv?",
                             file_format = "fastq",
                             type = "File",
                             status = "released",
                             fields = c("accession", "read_count", "md5sum",
                                        "controlled_by", "paired_end",
                                        "paired_with", "replicate", "target")) {
  query <- paste(list(paste0("type=", type),
                      paste0("status=", status),
                      paste0("file_format=", file_format),
                      paste0("dataset=%2Fexperiments%2F", experiment_accession, "%2F"),
                      map_chr(fields, ~paste0("field=", .))) %>%
                   flatten(),
                 collapse = "&")
  url <- paste0(base_url, query)
  return(url)
}
encode_file_info <- function(experiment_accession,
                             base_url = "https://www.encodeproject.org/report.tsv?",
                             file_format = "fastq",
                             type = "File",
                             status = "released",
                             fields = c("accession", "read_count", "md5sum",
                                        "controlled_by", "paired_end",
                                        "paired_with", "replicate", "target")) {
  path <- "report.tsv?"
  base_url <- modify_url("https://www.encodeproject.org/", path = path)
  url <- contstruct_query(experiment_accession,
                          base_url = base_url,
                          file_format,
                          type,
                          status,
                          fields)
  resp <- GET(url)
  if (http_error(resp)) {
    error_message <- content(resp, type = "text/html", encoding = "UTF-8") %>%
      xml_find_all("//p") %>%
      xml_text() %>%
      first()
    stop(
      sprintf(
        "ENCODE API request failed [%s]\n%s",
        status_code(resp),
        error_message
      ),
      call. = FALSE
    )
  }
  if (http_type(resp) != "text/tsv") {
    stop("API did not return text/tsv", call. = FALSE)
  }
  body <- read_tsv(content(resp, "text"), skip = 1) %>%
    clean_names()
  return(body)
}

get_fastq_info <- function(experiment_accession) {
  request <- GET(contstruct_query(experiment_accession)) 
  body <- read_tsv(content(request, "text"), skip = 1)
  return(body)
}
```

# Read in the experiment report downloaded from Encode
# we will only use the Accession column 
# Here we'll retrieve the read_count and md5 sum 
# as well as the file accession (with some other metadata)
# for the fastq files associated with each experiment.

```{r}
fastq_info <- read.table("experiment_report_2021_6_29_19h_49mfastq.tsv",
                        skip = 1, sep = "\t", header = T) %>%
  dplyr::select(Accession) %>%
  rename(experiment_accession = Accession) %>%
  distinct() %>%
  mutate(fastq_info = map(experiment_accession, ~encode_file_info(.))) %>%
  unnest(fastq_info) %>%
  rename(file_accession = accession) %>%
  clean_names()

  
write_csv(fastq_info, "design_input.csv")


md5_txt <- fastq_info %>%
  mutate(fastq_file = paste0(file_accession, ".fastq.gz")) %>%
  dplyr::select(md5sum, fastq_file)
write.table(md5_txt, "md5.txt", sep = "  ", quote = FALSE, row.names = FALSE, col.names = FALSE)
```



Awesome now we have a text file of encode file names and md5sum.
Let's run the md5sum test using -c to check a list and print the output to md5_status.txt

```{bash}
md5sum -c md5.txt 
#To output do the following in the directory with files to be tested
md5sum -c md5.txt > md5_status.txt
```




# We are getting some biosample information to maybe merge on later

```{r}
meta_data <- read.table("experiment_report_2021_6_29_19h_49mfastq.tsv",
                        skip = 1, sep = "\t", header = T)

meta_data_merge <- dplyr::select(meta_data, Controls, Accession, Biosample.summary, Biosample.term.name) %>%
  mutate(biosample = gsub(" ", "_", Biosample.summary), biosample = gsub("\\(|\\)", "", biosample)) %>%
  mutate(biosample_term = gsub(" ", "_", Biosample.term.name)) %>%
  dplyr::select(Controls, biosample, biosample_term, Accession) %>% 
  dplyr::rename(experiment_accession = Accession) 


fastq_info_merged <- merge(fastq_info, meta_data_merge)
```

## Starting to make design file
#TODO: check to see if the controls are comming down from encode API - it's not, so we'll grab them from the .tsv file in Controls column
```{r}
fastq_info_merged$fastq_1 <- paste0("../../data/chipseq_fastq/", fastq_info_merged$file_accession, ".fastq.gz")
fastq_info_merged$control <- gsub("/", "", gsub("/experiments/", "", fastq_info_merged$Controls))
fastq_info_merged$fastq_2 <- ""
fastq_info_merged$antibody <- gsub("-human", "", gsub("/", "", gsub("/targets/", "", fastq_info_merged$target)))

fastq_info_merged <- fastq_info_merged %>%
  unite(group, antibody, biosample, sep = ".", remove = F)


contrls <- fastq_info_merged %>% filter(Controls == "")

contrls$group <- contrls$experiment_accession

#Choose just one file for each input control. Exclude replicates by experiment_accession column 
contrls_unique <- contrls[!duplicated(contrls$experiment_accession), ]

samples <- fastq_info_merged %>% filter(Controls != "")
fastq_final <- rbind(contrls_unique, samples)

fastq_final <- fastq_final %>% 
  group_by(group) %>%
  mutate(replicate = 1:n())
design <- fastq_final %>%
  dplyr::select(group, replicate, fastq_1, fastq_2, antibody, control)

write_csv(design, "design.csv")

#I excluded some samples/files using excel
#antibody for controls have NA, correct that:

design_final <- read_csv("design_final.csv")

design_final$fastq_2 <- ""

design_final[is.na(design_final)] <- ""

#design_final$group <- gsub(",", "", design_final$group)

#design_final$fastq_1 <- gsub("../../","../", design_final$fastq_1)

design_final$group <- gsub("treated_with(.*)_Activin_A", "", design_final$group)

design_final$group <- gsub("Bone_morphogenetic_protein", "BMP", design_final$group)
design_final$group <- gsub("10_ng/mL", "", design_final$group)

#ENCSR554RPI: removed because it's private in ENCODE and we don;t have access.
design_final <- design_final %>% filter(control != "ENCSR554RPI", group != "ENCSR554RPI")

write_csv(design_final, "design_final.csv")


```


#Why keep just one control replicate?
The control column should be the group identifier for the controls for any given IP. The pipeline will automatically pair the inputs based on replicate identifier (i.e. where you have an equal number of replicates for your IP's and controls), alternatively, the first control sample in that group will be selected.

#Excluded samples/files: ENCSR842VGU
# also: ENCFF918EVM, ENCFF741BSF, ENCFF255JSS (failed, didnt match with replicates, same input control); ENCFF962AHN, ENCFF044YWU,ENCFF794RCZ, ENCFF523UYJ, ENCFF205RPZ, ENCFF435KUB, ENCFF632WAT, ENCFF254TVF (paired, read 2)
#change ENCFF886BMC (read2) for ENCFF633HXG (read1) at control ENCSR689CZH
#excluded ENCSR554RPI (input) and all its samples
