---
title: "figure_plots"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(ggplot2)
library(DESeq2)
library(VennDiagram)
library(ggdendro)
library(GenomicRanges)
library(RColorBrewer)
library(ggrepel)

my_pal <- brewer.pal(n = 11, name = 'RdBu')
```


```{r - color palette}
display.brewer.pal(n = 11, name = 'RdBu') #11 max colors in the palette

brewer.pal(n = 11, name = 'RdBu') #gives the color codes
#[1] (red) "#67001F" "#B2182B" "#D6604D" "#F4A582" "#FDDBC7" "#F7F7F7" "#D1E5F0" "#92C5DE" "#4393C3" #"#2166AC"
#[11] "#053061" (blue)
my_pal <- brewer.pal(n = 11, name = 'RdBu')

#use this in ggplot
+ scale_color_brewer(palette = "RdBu") +
```

For color palette https://www.datanovia.com/en/blog/the-a-z-of-rcolorbrewer-palette/

```{r - read in files}
#D10 vs D10 dox
deg10 <- read.csv("/scratch/Shares/rinn/isabela/CARDELL/rnaseq/resultadosDEtotalnorm_10P10Pdox_excel.csv")

#D15 vs D15dox
deg15 <- read.csv("/scratch/Shares/rinn/isabela/CARDELL/rnaseq/resultadosDEtotalnorm_15P15Pdox_excel.csv")

#D10 vs D15
deg_dif <- read.csv("/scratch/Shares/rinn/isabela/CARDELL/rnaseq/resultadosDEtotalnorm_10P15P_excel.csv")

#D10dox vs D15dox
deg_dox <- read.csv("/scratch/Shares/rinn/isabela/CARDELL/rnaseq/resultadosDEtotalnorm_10Pdox15Pdox_excel.csv")
```

```{r - apply filter for DEGs}
### Filter by FDR 0.05 and logFC 1.5

deg10_up <- deg10 %>%
  filter(padj <= 0.05 & log2FoldChange <= -1.5)
write_csv(deg10_up, "/scratch/Shares/rinn/isabela/CARDELL/rnaseq/deg10_up.csv")

deg10_up <- read.csv("/scratch/Shares/rinn/isabela/CARDELL/rnaseq/deg10_up.csv")

deg10_down <- deg10 %>%
  filter(padj <= 0.05 & log2FoldChange >= 1.5)
write_csv(deg10_down, "/scratch/Shares/rinn/isabela/CARDELL/rnaseq/deg10_down.csv")

deg10_down <- read.csv("/scratch/Shares/rinn/isabela/CARDELL/rnaseq/deg10_down.csv")

##

deg15_up <- deg15 %>%
  filter(padj <= 0.05 & log2FoldChange <= -1.5)
write_csv(deg15_up, "/scratch/Shares/rinn/isabela/CARDELL/rnaseq/deg15_up.csv")

deg15_up <- read.csv("/scratch/Shares/rinn/isabela/CARDELL/rnaseq/deg15_up.csv")

deg15_down <- deg15 %>%
  filter(padj <= 0.05 & log2FoldChange >= 1.5)
write_csv(deg15_down, "/scratch/Shares/rinn/isabela/CARDELL/rnaseq/deg15_down.csv")

deg15_down <- read.csv("/scratch/Shares/rinn/isabela/CARDELL/rnaseq/deg15_down.csv")

##

deg_dif_up <- deg_dif %>%
  filter(padj <= 0.05 & log2FoldChange <= -1.5)
write_csv(deg_dif_up, "/scratch/Shares/rinn/isabela/CARDELL/rnaseq/deg_dif_up.csv")

deg_dif_up <- read.csv("/scratch/Shares/rinn/isabela/CARDELL/rnaseq/deg_dif_up.csv")

deg_dif_down <- deg_dif %>%
  filter(padj <= 0.05 & log2FoldChange >= 1.5)
write_csv(deg_dif_down, "/scratch/Shares/rinn/isabela/CARDELL/rnaseq/deg_dif_down.csv")

deg_dif_down <- read.csv("/scratch/Shares/rinn/isabela/CARDELL/rnaseq/deg_dif_down.csv")

##

deg_dox_up <- deg_dox %>%
  filter(padj <= 0.05 & log2FoldChange <= -1.5)
write_csv(deg_dox_up, "/scratch/Shares/rinn/isabela/CARDELL/rnaseq/deg_dox_up.csv")

deg_dox_up <- read.csv("/scratch/Shares/rinn/isabela/CARDELL/rnaseq/deg_dox_up.csv")

deg_dox_down <- deg_dox %>%
  filter(padj <= 0.05 & log2FoldChange >= 1.5)
write_csv(deg_dox_down, "/scratch/Shares/rinn/isabela/CARDELL/rnaseq/deg_dox_down.csv")

deg_dox_down <- read.csv("/scratch/Shares/rinn/isabela/CARDELL/rnaseq/deg_dox_down.csv")

```

```{r - volcano plot}

#D10

deg10_volcano <- deg10 %>%
  select(Row.names, log2FoldChange, padj, external_gene_name)

#clean NA from padj
deg10_volcano <- deg10_volcano[complete.cases(deg10_volcano), ]

#correct (-) of logFC == (-) will mean DOWN in dox
deg10_volcano$log2FoldChange <- (deg10_volcano$log2FoldChange*-1)

#assign UP & DOWN
deg10_volcano$deg <- "NO" #set non-deg first
deg10_volcano$deg[deg10_volcano$log2FoldChange >= 1.5 & deg10_volcano$padj <= 0.05] <- "UP"
deg10_volcano$deg[deg10_volcano$log2FoldChange <= -1.5 & deg10_volcano$padj <= 0.05] <- "DOWN"

#label for top 10 genes names
top_padj <- deg10_volcano %>%
  filter(padj <= 0.05)
top_up <- top_padj[with(top_padj,order(-log2FoldChange)),]
top_up <- top_up[1:10,]

top_down <- top_padj[with(top_padj,order(log2FoldChange)),]
top_down <- top_down[1:10,]

top_names <- rbind(top_down, top_up)
top_names <- top_names %>%
  mutate(label = top_names$external_gene_name)

deg10_volcano <- deg10_volcano %>%
  left_join(top_names)

#VOLCANO PLOT = log2GC & -log10(padj)

ggplot(deg10_volcano, aes(x=log2FoldChange, y=-log10(padj), col=deg, label=label)) + 
  geom_point() +
  #theme_minimal() +
  scale_color_manual(values=c("#053061", "black", "#67001F")) + #color assigned in alphabetical order (use same as heatmap = brewer palette RdBu)
  #geom_text_repel() + #this add names of top10 genes
  theme(text = element_text(size = 12), axis.text = element_text(size = 12))  

#save as pdf landscape 4 x 3.5



#D15 

deg15_volcano <- deg15 %>%
  select(Row.names, log2FoldChange, padj, external_gene_name)

#clean NA from padj
deg15_volcano <- deg15_volcano[complete.cases(deg15_volcano), ]

#correct (-) of logFC == (-) will mean DOWN in dox
deg15_volcano$log2FoldChange <- (deg15_volcano$log2FoldChange*-1)

#assign UP & DOWN
deg15_volcano$deg <- "NO" #set non-deg first
deg15_volcano$deg[deg15_volcano$log2FoldChange >= 1.5 & deg15_volcano$padj <= 0.05] <- "UP"
deg15_volcano$deg[deg15_volcano$log2FoldChange <= -1.5 & deg15_volcano$padj <= 0.05] <- "DOWN"

#label for top 10 genes names
top_padj <- deg15_volcano %>%
  filter(padj <= 0.05)
top_up <- top_padj[with(top_padj,order(-log2FoldChange)),]
top_up <- top_up[1:10,]

top_down <- top_padj[with(top_padj,order(log2FoldChange)),]
top_down <- top_down[1:10,]

top_names <- rbind(top_down, top_up)
top_names <- top_names %>%
  mutate(label = top_names$external_gene_name)

deg15_volcano <- deg15_volcano %>%
  left_join(top_names)

#VOLCANO PLOT = log2GC & -log10(padj)

ggplot(deg15_volcano, aes(x=log2FoldChange, y=-log10(padj), col=deg, label=label)) + 
  geom_point() +
  #theme_minimal() +
  scale_color_manual(values=c("#053061", "black", "#67001F")) + #color assigned in alphabetical order (use same as heatmap = brewer palette RdBu)
  #geom_text_repel() + #this add names of top10 genes
  theme(text = element_text(size = 12), axis.text = element_text(size = 12))  

#save as pdf landscape 4 x 3.5

#done
```

```{r - GOs}
#from Gprofiler - top 5 each

#BP

#D10

bp10up <- read.csv("/scratch/Shares/rinn/isabela/CARDELL/rnaseq/gProfiler_hsapiens_7-2-2022_1-19-29 PM__intersections_deg10_up.csv")
bp10up_plot <- bp10up %>%
  filter(source == "GO:BP") %>%
  head(n=5) %>%
  mutate(freq = intersection_size/term_size*100)

bp10down <- read.csv("/scratch/Shares/rinn/isabela/CARDELL/rnaseq/gProfiler_hsapiens_7-2-2022_1-51-38 PM__intersections_deg10_down.csv")
bp10down_plot <- bp10down %>%
  filter(source == "GO:BP") %>%
  head(n=5) %>%
  mutate(freq = intersection_size/term_size*100)

ggplot(bp10up_plot, aes(x = negative_log10_of_adjusted_p_value , y = term_name )) +
  geom_bar(stat = "identity", fill = "#67001F", width=0.6, position = position_dodge(width = 0)) +
  theme(text = element_text(size = 12), axis.text = element_text(size = 12), axis.text.x = element_text(angle = 360)) +
  xlab("-log10(padj)")
  #save as pdf landscap 3.5 x 4.5

ggplot(bp10down_plot, aes(x = negative_log10_of_adjusted_p_value , y = reorder(term_name, negative_log10_of_adjusted_p_value))) +
  geom_bar(stat = "identity", fill = "#053061", width=0.6, position = position_dodge(width = 0)) +
  theme(text = element_text(size = 12), axis.text = element_text(size = 12), axis.text.x =    element_text(angle = 360)) +
  xlab("-log10(padj)")
  #save as pdf landscap 3.5 x 4.5


#D15

bp15up <- read.csv("/scratch/Shares/rinn/isabela/CARDELL/rnaseq/gProfiler_hsapiens_7-2-2022_2-16-48 PM__intersections_d15_up.csv")
bp15up_plot <- bp15up %>%
  filter(source == "GO:BP") %>%
  head(n=6) %>%
  mutate(freq = intersection_size/term_size*100)
bp15up_plot <- bp15up_plot[-c(5), ] #keep heart development and exclude striated muscle tissue development

bp15down <- read.csv("/scratch/Shares/rinn/isabela/CARDELL/rnaseq/gProfiler_hsapiens_7-2-2022_2-18-51 PM__intersections_deg15_down.csv")
bp15down_plot <- bp15down %>%
  filter(source == "GO:BP") %>%
  head(n=5) %>%
  mutate(freq = intersection_size/term_size*100)

ggplot(bp15up_plot, aes(x = negative_log10_of_adjusted_p_value , y = reorder(term_name, negative_log10_of_adjusted_p_value))) +
  geom_bar(stat = "identity", fill = "#67001F", width=0.6, position = position_dodge(width = 0)) +
  theme(text = element_text(size = 12), axis.text = element_text(size = 12), axis.text.x = element_text(angle = 360)) +
  xlab("-log10(padj)")
  #save as pdf landscap 3.5 x 4.5

ggplot(bp15down_plot, aes(x = negative_log10_of_adjusted_p_value , y = reorder(term_name, negative_log10_of_adjusted_p_value))) +
  geom_bar(stat = "identity", fill = "#053061", width=0.6, position = position_dodge(width = 0)) +
  theme(text = element_text(size = 12), axis.text = element_text(size = 12), axis.text.x = element_text(angle = 360)) +
  xlab("-log10(padj)")
  #save as pdf landscap 3.5 x 4.5

```

```{r - final(?) GO filters for heatmap}

#filtered manually BP terms related to cardio
bp15up_filter <- read_csv("/scratch/Shares/rinn/isabela/CARDELL/rnaseq/gProfiler_hsapiens_7-2-2022_2-16-48 PM__intersections_d15_up_cardiofilter.csv")

#filtered manually molecular functions: structural or ion related
mol15up_filter <- read_csv("/scratch/Shares/rinn/isabela/CARDELL/rnaseq/gProfiler_hsapiens_7-2-2022_2-16-48 PM__intersections_d15_up_mol_function.csv")


cm_dif <- bp15up_filter %>%
  filter(term_name == "cardiocyte differentiation") %>%
  select(genes, term_name)

struc_muscle <- mol15up_filter %>%
  filter(term_name == "structural constituent of muscle") %>%
  select(genes, term_name)

heart_cont <- bp15up_filter %>%
  filter(term_name == "heart contraction") %>%
  select(genes, term_name)
#ion15up_genes

go_genes_final <- rbind(cm_dif, struc_muscle, heart_cont)

go_genes_final <- go_genes_final %>%
  dplyr :: rename(external_gene_name = genes) %>%
  left_join(gene_description)

write_csv(go_genes_final, "/scratch/Shares/rinn/isabela/CARDELL/rnaseq/go_genes_final.csv")

#after manual selection

go_genes_heatmap <- read_csv("/scratch/Shares/rinn/isabela/CARDELL/rnaseq/go_genes_final_manual_curated.csv")
```

```{r - genes heatmap}
heatmap_cm <- go_genes_heatmap %>%
  filter(category == "cardiocyte differentiation") %>%
  select(external_gene_name)

heatmap_struc <- go_genes_heatmap %>%
  filter(category == "structural constituent of muscle") %>%
  select(external_gene_name)

heatmap_heart <- go_genes_heatmap %>%
  filter(category == "heart contraction") %>%
  select(external_gene_name)


#grab COUNTS (???see what we plot here for real ############)
deg15_counts <- deg15 %>%
  select(external_gene_name, starts_with("X")) %>% select(-X)

#merge counts in each gene list & calculate mean counts

heatmap_cm <- heatmap_cm %>%
  left_join(deg15_counts) %>%
  pivot_longer(!external_gene_name, names_to = "sample", values_to = "counts")
heatmap_cm_counts <- heatmap_cm %>%
  mutate(day = gsub("X[1-3]", "", heatmap_cm$sample)) %>%
  group_by(external_gene_name, day) %>%
  mutate(mean_count = mean(counts)) %>%
  select(external_gene_name, day, mean_count) %>%
  distinct() %>%
  pivot_wider(names_from = "day", values_from = "mean_count") %>%
  column_to_rownames(var = "external_gene_name")


heatmap_struc <- heatmap_struc %>%
  left_join(deg15_counts) %>%
  pivot_longer(!external_gene_name, names_to = "sample", values_to = "counts")
heatmap_struc_counts <- heatmap_struc %>%
  mutate(day = gsub("X[1-3]", "", heatmap_struc$sample)) %>%
  group_by(external_gene_name, day) %>%
  mutate(mean_count = mean(counts)) %>%
  select(external_gene_name, day, mean_count) %>%
  distinct() %>%
  pivot_wider(names_from = "day", values_from = "mean_count") %>%
  column_to_rownames(var = "external_gene_name")

heatmap_heart <- heatmap_heart %>%
  left_join(deg15_counts) %>%
  pivot_longer(!external_gene_name, names_to = "sample", values_to = "counts")
heatmap_heart_counts <- heatmap_heart %>%
  mutate(day = gsub("X[1-3]", "", heatmap_heart$sample)) %>%
  group_by(external_gene_name, day) %>%
  mutate(mean_count = mean(counts)) %>%
  select(external_gene_name, day, mean_count) %>%
  distinct() %>%
  pivot_wider(names_from = "day", values_from = "mean_count") %>%
  column_to_rownames(var = "external_gene_name")


## PLOT

#transform log
heatmap_cm_counts_log <- log2(heatmap_cm_counts)

heatmap_struc_counts_log <- log2(heatmap_struc_counts)

heatmap_heart_counts_log <- log2(heatmap_heart_counts)

## Scale & adjust dendrogram (The most similar columns will appear clustered toward the left side of the plot. The columns that are more distant from each other will appear clustered toward the right side of the plot)
#function
sort_hclust <- function(...) as.hclust(dendsort(as.dendrogram(...)))

#cardiocyte differentiation
heatmap_cm_counts_log_scaled <- t(scale(t(heatmap_cm_counts_log))) 
scaled_cluster_cols <- hclust(dist(t(heatmap_cm_counts_log_scaled)))
scaled_cluster_cols <- sort_hclust(scaled_cluster_cols)

pheatmap::pheatmap(heatmap_cm_counts_log_scaled, show_rownames = TRUE, 
                   cluster_rows = T, cluster_cols = scaled_cluster_cols,
                   breaks = seq(-1.2, 1.2, length.out = length(my_pal)),
                   color = rev(my_pal), border_color = NA, fontsize_row = 10, fontsize_col = 10)

#structural constituent of muscle
heatmap_struc_counts_log_scaled <- t(scale(t(heatmap_struc_counts_log))) 
scaled_cluster_cols <- hclust(dist(t(heatmap_struc_counts_log_scaled)))
scaled_cluster_cols <- sort_hclust(scaled_cluster_cols)

pheatmap::pheatmap(heatmap_struc_counts_log_scaled, show_rownames = TRUE, 
                   cluster_rows = T, cluster_cols = scaled_cluster_cols,
                   breaks = seq(-1.2, 1.2, length.out = length(my_pal)),
                   color = rev(my_pal), border_color = NA, fontsize_row = 10, fontsize_col = 10)


#heart contraction
heatmap_heart_counts_log_scaled <- t(scale(t(heatmap_heart_counts_log))) 
scaled_cluster_cols <- hclust(dist(t(heatmap_heart_counts_log_scaled)))
scaled_cluster_cols <- sort_hclust(scaled_cluster_cols)

pheatmap::pheatmap(heatmap_heart_counts_log_scaled, show_rownames = TRUE, 
                   cluster_rows = T, cluster_cols = scaled_cluster_cols,
                   breaks = seq(-1.2, 1.2, length.out = length(my_pal)),
                   color = rev(my_pal),
                   border_color = NA, fontsize_row = 10, fontsize_col = 10)


# Save as pdf
# Landscape 5 x 2.94
```
