---
title: "rnaseq_analysis"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(ggplot2)
library(DESeq2)
library(VennDiagram)
library(ggdendro)
library(GenomicRanges)
source("/scratch/Shares/rinn/isabela/rna_protein_complexes/analysis/util/_plot_theme.R")
```


```{r - read in files}
#D10 vs D10 dox
deg10 <- read.csv("/scratch/Shares/rinn/isabela/CARDELL/rnaseq/resultadosDEtotalnorm_10P10Pdox_excel.csv")

#D15 vs D15dox
deg15 <- read.csv("/scratch/Shares/rinn/isabela/CARDELL/rnaseq/resultadosDEtotalnorm_15P15Pdox_excel.csv")

#D10 vs D15
deg_dif <- read.csv("/scratch/Shares/rinn/isabela/CARDELL/rnaseq/resultadosDEtotalnorm_10P15P_excel.csv")

#D10dox vs D15dox
deg_dox <- read.csv("/scratch/Shares/rinn/isabela/CARDELL/rnaseq/resultadosDEtotalnorm_10Pdox15Pdox_excel.csv")
```

```{r - apply filter for DEGs}
### Filter by FDR 0.05 and logFC 1.5

deg10_up <- deg10 %>%
  filter(padj <= 0.05 & log2FoldChange <= -1.5)
write_csv(deg10_up, "/scratch/Shares/rinn/isabela/CARDELL/rnaseq/deg10_up.csv")

deg10_down <- deg10 %>%
  filter(padj <= 0.05 & log2FoldChange >= 1.5)
write_csv(deg10_down, "/scratch/Shares/rinn/isabela/CARDELL/rnaseq/deg10_down.csv")

##

deg15_up <- deg15 %>%
  filter(padj <= 0.05 & log2FoldChange <= -1.5)
write_csv(deg15_up, "/scratch/Shares/rinn/isabela/CARDELL/rnaseq/deg15_up.csv")

deg15_down <- deg15 %>%
  filter(padj <= 0.05 & log2FoldChange >= 1.5)
write_csv(deg15_down, "/scratch/Shares/rinn/isabela/CARDELL/rnaseq/deg15_down.csv")

##

deg_dif_up <- deg_dif %>%
  filter(padj <= 0.05 & log2FoldChange <= -1.5)
write_csv(deg_dif_up, "/scratch/Shares/rinn/isabela/CARDELL/rnaseq/deg_dif_up.csv")

deg_dif_down <- deg_dif %>%
  filter(padj <= 0.05 & log2FoldChange >= 1.5)
write_csv(deg_dif_down, "/scratch/Shares/rinn/isabela/CARDELL/rnaseq/deg_dif_down.csv")

##

deg_dox_up <- deg_dox %>%
  filter(padj <= 0.05 & log2FoldChange <= -1.5)
write_csv(deg_dox_up, "/scratch/Shares/rinn/isabela/CARDELL/rnaseq/deg_dox_up.csv")

deg_dox_down <- deg_dox %>%
  filter(padj <= 0.05 & log2FoldChange >= 1.5)
write_csv(deg_dox_down, "/scratch/Shares/rinn/isabela/CARDELL/rnaseq/deg_dox_down.csv")

```

```{r - how much CARDEL affect overall dif}
#Compare if what changes between D10 and D15 in regular diff is the same as with dox
#UP: 914 & 1086
cardel_effect_overall <- intersect(deg_dif_up$Row.names, deg_dox_up$Row.names)
#537

#find what don't intersect
deg_dox_up$only_dox <- ifelse(deg_dox_up$Row.names %in% deg_dif_up$Row.names, "TRUE", "dox")
only_dox <- deg_dox_up %>%
  filter(only_dox == "dox") #549
write_csv(only_dox, "/scratch/Shares/rinn/isabela/CARDELL/rnaseq/only_dox.csv")

common_dif <- deg_dox_up %>%
  filter(only_dox == "TRUE") #537
write_csv(common_dif, "/scratch/Shares/rinn/isabela/CARDELL/rnaseq/common_dif.csv")

#Let's see how much the only_dox intersect with D15 x D15dox
#549 & 393
intersect2 <- intersect(only_dox$Row.names, deg15_up$Row.names)
#147

only_dox$D15 <- ifelse(only_dox$Row.names %in% deg15_up$Row.names, "common", "not")
only_dox_D15 <- only_dox %>%
  filter(D15 == "not") 
#402 genes that CARDEL also helps up-express between D10 and D15 that didn't show up in the D15 comparison
write_csv(only_dox_D15, "/scratch/Shares/rinn/isabela/CARDELL/rnaseq/only_dox_D15_not.csv")

#DOWN: 364 & 351
cardel_effect_overall_d <- intersect(deg_dif_down$Row.names, deg_dox_down$Row.names)
#122

##TODO: most of genes not similar - what are they and how they express??

```

```{r - D10dox is more similar to D15?}
#What CARDEL helps to express earlier in D10?
#Compare D10xD15 to D10xD10dox
#UP: 914 & 195
intersection <- intersect(deg_dif_up$Row.names, deg10_up$Row.names)
#34
intersection <- as.data.frame(intersection)
colnames(intersection) <- c("Row.names")
intersection <- intersection %>% left_join(deg_dif_up)
genes <- intersection %>% select(external_gene_name)
write.csv(genes,"/scratch/Shares/rinn/isabela/CARDELL/rnaseq/genes.csv")

#GO gprofiler: metal ion transport for the 34 genes
```


```{r - template####}
pdf("/scratch/Shares/rinn/isabela/rna_protein_complexes/analysis/00_mESC_grad/analysis/01_general_properties/results/ven_genes_filtered_tpm_old_newdata.pdf")

venn_plot <- draw.pairwise.venn(length(genes_old$gene_id),
                                length(filter_both$gene_id),
                                length(gene_intersect$gene_intersect),
                                category = c("Cody","Isabela"),
                                scaled=TRUE,
                                fill = c("dark gray", "red"),
                                alpha = rep(0.5,2),
                                cat.pos = c(-30,30),
                                lty = "blank",
                                cex = 2,
                                ext.text = TRUE,
                                cat.dist = c(0.03, 0.03),
                                cat.cex = 2)
dev.off()
```

```{r - clustering}
#make a file with norm_counts (#TODO: ask WHAT ARE THESE NUMBERS!!!!!!)
norm_counts <- deg10 %>%
  select(Row.names, starts_with("X")) %>%
  select(!X)

norm_counts <- distinct(norm_counts)

norm_counts <- norm_counts %>%
  column_to_rownames(var = "Row.names")

#scale to plot
norm_counts_scaled <- t(scale(t(norm_counts))) 

pheatmap::pheatmap(norm_counts_scaled, show_rownames = FALSE, 
                   cluster_rows = F, cluster_cols = T,
                   breaks = seq(-3, 3, length.out = length(col_pal10)),
                   color = col_pal10, border_color = NA, fontsize_row = 8)
```
#TODO: cluster with a cardiac specific list of genes (maybe d10dox up??)
